### ISSUE TITLE
Nordic PCA10059 support without additional Segger debugger

### ISSUE THREAD

**Issue by larsgk** [zephyrproject-rtos/zephyr#9055]
A collection of pull requests (and issues) related to PCA10059 (Dongle) suport:

- [x] Add add nrf52840_pca10059 board support ( #8812 )
- [x] pawelzadrozniak#1 to be merged to #8787 .
- [x] DNM Add nRF52840 USBD driver ( #8787 )
- [x] MCUboot Erase flash pages progressively (https://github.com/runtimeco/mcuboot/pull/323)
- [x] MCUboot nrf52840_pca10059 support ( https://github.com/runtimeco/mcuboot/pull/318 )
- [x] Document how to use nrfutil to flash (https://github.com/zephyrproject-rtos/zephyr/pull/11002)

/cc @lemrey @nvlsianpu

**Comment by carlescufi** [comment]
@larsgk thanks for creating this.

**Comment by carlescufi** [comment]
cc @jhedberg

**Comment by lemrey** [comment]
https://github.com/pawelzadrozniak/zephyr/pull/1 to be merged to #8787 .

**Comment by postlund** [comment]
I'm stuck with a few of these dongles that I want to flash with zephyr, but I don't have an external debugger (yet). If I understand it correctly, once this is done, I can flash without an additional debugger but I would still need one in the first place to place mcu_boot on the target, right? Using DFU-flash is not sufficient?

**Comment by carlescufi** [comment]
@postlund there is an issue with the current USB driver that @pawelzadrozniak. I think then that @lemrey needs to update the documentation on the board in order to describe how to use Zephyr without an external debugger.

**Comment by postlund** [comment]
Perfect, thank you!

**Comment by carlescufi** [comment]
@postlund @larsgk can you retry with current master? a fix was just merged https://github.com/zephyrproject-rtos/zephyr/pull/10934

**Comment by carlescufi** [comment]
@lemrey it looks like all the features are in place for this now. We just need to document how to transition from the built-in bootloader to MCUboot in order for people to be able to use this without a debugger.

**Comment by larsgk** [comment]
I can try when I get home

**Comment by tugstugi** [comment]
Is there now a documentation how to replace the stock bootloader without jlink?

**Comment by carlescufi** [comment]
@tugstugi not yet, @lemrey is working on it and will report soon

**Comment by Xenoamor** [comment]
![image](https://user-images.githubusercontent.com/7109951/48707281-d2727a00-ebf6-11e8-97f3-7223c86b9c7e.png)
I've tried uploading the latest firmware via j-link but I keep seeing this. It doesn't appear to be working on Linux either
Any idea what might be happening?

**Comment by Xenoamor** [comment]
![image](https://user-images.githubusercontent.com/7109951/48710015-149fb980-ebff-11e8-9288-09d84154cdb9.png)
Any idea why I can't get the hci_usb sample to work on this board? It appears to fail in the same way on both windows and linux. I'm uploading to the board via j-link and the blinky example works via this method

**Comment by lemrey** [comment]
Hi @Xenoamor, on this board the default peripheral used for HCI over UART is a UARTE peripheral. At the moment we are fixing an issue in the UARTE driver which is preventing the hci_uart sample to work as intented. Could you please check if it works any better with this patch https://github.com/zephyrproject-rtos/zephyr/pull/11475 ?

**Comment by Xenoamor** [comment]
Thanks, I'll give it a try when I'm back in the office tomorrow! Just to clarify that this is an issue I'm seeing with hci_usb and not hci_uart though

**Comment by carlescufi** [comment]
@Xenoamor so this is a completely different issue, not related to using this board without a debugger? Are you flashing using J-Link?

**Comment by Xenoamor** [comment]
Sorry yes I'm using a J-Link and nrfjprog. Should I make a new issue for this?

**Comment by carlescufi** [comment]
@Xenoamor yes please, this is completely unrelated.

**Comment by lemrey** [comment]
> Thanks, I'll give it a try when I'm back in the office tomorrow! Just to clarify that this is an issue I'm seeing with hci_usb and not hci_uart though

Sorry @Xenoamor, I had misunderstood. What I said applies for HCI over UARTE.
Follow carlescufi's advice then and create a new issue.

**Comment by carlescufi** [comment]
All the required PRs have now been merged.
Thanks!
